src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
@pierlo-upitup Yeah! Let's embrace the groups endpoint to the fullest 🤘 I like the idea of having a tree like structure, which is even better than a flat list with all items!

I wonder if we could split this refactoring into two parts to maintain a manageable PR size:

1. Needed to properly support empty groups: Provide a `Groups/AppsStore` `items` (Object including groups and apps), `apps` (flat list including all apps) and `groups` (flat list including all groups) getter. Remove the respective duplication and obsolete pieces (e.g. `getGroupsFromApps`) from the `AppListComponent ` and rename the component and Store as appropriate.

2. Improve structure and maintainability: Refactor components and stores to use `GroupsStore.items` and remove `groupItems` and the likes if applicable.

What do you think?